src/main.rs

```diff
@@ -358,7 +364,20 @@ async fn forward_request(
         })
         .collect::<Vec<_>>();
 
-    let entire_body = body::to_bytes(body).await?.to_vec();
+    // Limit request body size
+    let body = Limited::new(body, REQUEST_BODY_SIZE_LIMIT);
+    let entire_body = match hyper::body::to_bytes(body).await {
+        Ok(data) => data,
+        Err(err) => {
+            if err.downcast_ref::<LengthLimitError>().is_some() {
+                return Ok(Response::builder()
+                    .status(StatusCode::PAYLOAD_TOO_LARGE)
+                    .body(Body::from("Request size exceeds limit"))?);
+            }
+            return Err(err);
+        }
+    }
+    .to_vec();
 
     slog::trace!(logger, "<<");
```

Comment:
That wasn't part of my change, but I think in the near future we'd like to refactor a lot of this code base and extract most of the logging out into it's own middlewares, etc. I.e the core handlers should not know anything about logging or tracing or metrics. Even further, the core handlers won't know anything about request size limiting either! It will all be done in small self-contained middlewares.

Reply:
I would advise against extracting this in this PR. Should be a separate PR.